src/EditorFeatures/Core/InlineRename/InlineRenameSession.cs

```diff
@@ -293,20 +292,31 @@ private void OnSubjectBuffersConnected(object sender, SubjectBuffersConnectedEve
             }
         }
 
-        private void UpdateReferenceLocationsTask(JoinableTask<IInlineRenameLocationSet> findRenameLocationsTask)
+        private void UpdateReferenceLocationsTask()
         {
             _threadingContext.ThrowIfNotOnUIThread();
 
             var asyncToken = _asyncListener.BeginAsyncOperation("UpdateReferencesTask");
+
+            var currentOptions = _options;
+            var currentRenameLocationsTask = _allRenameLocationsTask;
+            var cancellationToken = _cancellationTokenSource.Token;
+
             _allRenameLocationsTask = _threadingContext.JoinableTaskFactory.RunAsync(async () =>
             {
-                var inlineRenameLocations = await findRenameLocationsTask.JoinAsync().ConfigureAwaitRunInline();
+                // Join prior work before proceeding, since it performs a required state update.
+                // https://github.com/dotnet/roslyn/pull/34254#discussion_r267024593
+                if (currentRenameLocationsTask != null)
+                    await _allRenameLocationsTask.JoinAsync(cancellationToken).ConfigureAwait(false);
+
+                await TaskScheduler.Default;
```

Comment:
Do we still need this await here at all?

Reply:
no.  but it doesn't hurt.  it's an explicitly nice way of saying: ensure we're off hte ui thread.